integration/mediation-tests/tests-transport/src/test/java/org/wso2/carbon/esb/jms/inbound/transport/test/JMSInboundMessageSupportTestCase.java

```diff
@@ -0,0 +1,160 @@
+    }
+
+    @Test(
+            groups = { "wso2.esb" },
+            description = "JMS Inbound Endpoint Polling BytesMessages from a Queue"
+    )
+    public void testBytesMessageType() throws Exception {
+        endpointName = "BytesMessageEndpoint";
+        queueName = "BytesMessageQueue";
+        logViewerClient.clearLogs();
+        try {
+            jmsQueueMessageProducer.connect(queueName);
+            for (int i = 0; i < numberOfMessages; i++) {
+                jmsQueueMessageProducer.sendBytesMessage(("<?xml version='1.0' encoding='UTF-8'?>"
+                        + "<soapenv:Envelope xmlns:soapenv=\"http://schemas.xmlsoap.org/soap/envelope/\""
+                        + " xmlns:ser=\"http://services.samples\" xmlns:xsd=\"http://services.samples/xsd\">"
+                        + "   <soapenv:Header/>"
+                        + "   <soapenv:Body>"
+                        + "      <ser:placeOrder>"
+                        + "         <ser:order>"
+                        + "            <xsd:price>100</xsd:price>"
+                        + "            <xsd:quantity>2000</xsd:quantity>"
+                        + "            <xsd:symbol>JMSBytes</xsd:symbol>"
+                        + "         </ser:order>"
+                        + "      </ser:placeOrder>"
+                        + "   </soapenv:Body>"
+                        + "</soapenv:Envelope>").getBytes());
+                log.info("BytesMessage " + i + " published to the JMS Queue");
+            }
+        } finally {
+            jmsQueueMessageProducer.disconnect();
+        }
+
+        addInboundEndpoint(getEndpointConfig(endpointName, queueName));
+        Thread.sleep(10000);
+        int numberOfMessagesReceived = 0;
+        logEvents = logViewerClient.getAllRemoteSystemLogs();
+        for (LogEvent logEvent : logEvents) {
+            if (logEvent.getMessage().contains("<xsd:symbol>JMSBytes</xsd:symbol>")) {
+                numberOfMessagesReceived++;
+            }
+        }
+        deleteInboundEndpointFromName(endpointName);
+        Assert.assertTrue(numberOfMessagesReceived == numberOfMessages, "Couldn't consume BytesMessages from Queue");
+    }
+
+    @Test(
+            groups = { "wso2.esb" },
+            description = "JMS Inbound Endpoint Polling MapMessages from a Queue"
+    )
+    public void testMapMessageType() throws Exception {
+        endpointName = "MapMessageEndpoint";
+        queueName = "MapMessageQueue";
+        logViewerClient.clearLogs();
+        try {
+            jmsQueueMessageProducer.connect(queueName);
+            for (int i = 0; i < 3; i++) {
+                jmsQueueMessageProducer.sendMapMessage();
+                log.info("MapMessage " + i + " published to the JMS Queue");
+            }
+        } finally {
+            jmsQueueMessageProducer.disconnect();
+        }
+
+        addInboundEndpoint(getEndpointConfig(endpointName, queueName));
+        Thread.sleep(10000);
```

Comment:
@nuwanw  @sdkottegoda @a5anka @pamod @nandika @isudana one day ( near future) we need to get rid of thread.sleeps with a proper mechanism.

Reply:
Instead of having for Thread.sleep(10000), here we can use a loop, and in each iteration wait for 1000 milliseconds until maximum 10000 milliseconds.

Reply:
I wrote a Utils method for that https://github.com/wso2/product-ei/blob/master/integration/mediation-tests/tests-common/integration-test-utils/src/main/java/org/wso2/esb/integration/common/utils/Utils.java#L190